whittle/models/gpt/extract.py

```diff
@@ -25,15 +26,33 @@ def extract_sub_network(model, sub_network_config):
         sub_network_block.attn.proj.load_state_dict(state_dict)
 
         # MLP
-        state_dict = extract_linear(block.mlp.fc)
-        sub_network_block.mlp.fc.load_state_dict(state_dict)
-
-        state_dict = extract_linear(block.mlp.proj)
-        sub_network_block.mlp.proj.load_state_dict(state_dict)
+        extract_mlp(block.mlp, sub_network_block.mlp)
 
     return sub_network
 
 
+def extract_mlp(mlp, sub_mlp):
+    if isinstance(mlp, GptNeoxMLP):
+        state_dict = extract_linear(mlp.fc)
+        sub_mlp.fc.load_state_dict(state_dict)
+
+        state_dict = extract_linear(mlp.proj)
+        sub_mlp.proj.load_state_dict(state_dict)
+    elif isinstance(mlp, LLaMAMLP):
+        state_dict = extract_linear(mlp.fc_1)
+        sub_mlp.fc_1.load_state_dict(state_dict)
+
+        state_dict = extract_linear(mlp.fc_2)
+        sub_mlp.fc_2.load_state_dict(state_dict)
+
+        state_dict = extract_linear(mlp.proj)
+        sub_mlp.proj.load_state_dict(state_dict)
+    else:
+        raise ValueError(
+            "Cannot extract MLP, supported MLP classes are GptNeoxMLP and LLaMAMLP."
+        )
```

Comment:
Could we support gemma-mlp too here @gabikadlecova ?

Reply:
It is already supported since GemmaMLP is derived from LLamaMLP
https://github.com/whittle-org/whittle/blob/52b0348b013bab9e1af5ffc804b4d7da45145fa0/whittle/models/gpt/blocks/mlp.py#L82

I'll add it to the ValueError message to make it more clear